internal/model/globalyaml.go

```diff
@@ -52,3 +52,14 @@ func (YAMLManifest) DockerRef() reference.Named {
 
 	return n
 }
+
+func (y YAMLManifest) Empty() bool {
+	return y.k8sYAML == ""
+}
+
+func (y1 YAMLManifest) Equal(y2 YAMLManifest) bool {
```

Comment:
it would probably be better for GlobalYAMLBuildController to do its own equality check with the state that it's copying

Reply:
Like [this](https://github.com/windmilleng/tilt/pull/647/commits/199292617d04f9f070d6ffeaf551da9486d839bd)?

Reply:
no, i mean it compares the individual fields it cares about

Reply:
the upshot is that this Equality function is not good. 

you even acknowledge that it's not good in the TODO.

the reason it's not good is because it treats this equality function as a general equality function. But in fact it's an equality function that's very narrowly tailored to GlobalYAMLBuildController's needs

Reply:
yeah, good point :)